docs/contributing/code-quality-principles.md

```diff
@@ -0,0 +1,173 @@
+# Code Quality Principles
+
+We strictly follow [coding standards](coding-standards.md) already and we are committing ourselves to improve Shopsys code quality behind normal coding standards.
+We decided to write down principles that improve code quality so the code quality is consistent between team members.
+Due to the written form, we have a place to point during code review to cut down discussions.
+
+These principles are not the only we keep and follow, these are a couple of important ones that we agreed in the team and that are unlikely to be violated.
+We may add, change or remove principles as we improve our practices, so this more a living document than a definitive guide.
+
+### When do we follow principles
+
+We follow principles during writing **new code**, during **changing current code** and we check principles during **code review**.
+
+You can find an old code that doesn't follow these principles and we are ok with that as we are not able to refactor the whole framework at once.
+It may also happen that changed code cannot follow a principle because it has to be [backward compatible](backward-compatibility-promise.md),
+backward compatibility is more important than the code quality (it may sound sad, but it is necessary).
+
+These principles are not rules, they can be violated when it has a reason.
+The reason has to be discussed with the reviewer and may be written in the commit message.
+
+## Principles
+
+* [Dependency injection](#dependency-injection)
```

Comment:
Shuldnt these links scroll to the part of the documentation? Because it aint doin anything for me.. Maybe Mac stuff?

Reply:
you are right, they should...
works on my machine™

Reply:
Okay, works out of a diff